src/plugin_runtime/kernel.rs

```diff
@@ -101,17 +89,25 @@ impl Kernel {
 
         Ok(())
     }
+
+    pub fn get_plugins(&self) -> &[Plugin] {
+        &self.plugins[..]
+    }
```

Comment:
Can it be called just `plugins`? Using `get_` prefix is disgouraged by [Rust API guidelines](https://rust-lang-nursery.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter).